pkg/process/procutil/process_linux.go

```diff
@@ -583,6 +583,40 @@ func (p *Probe) getFDCount(pidPath string) int32 {
 	return int32(len(names))
 }
 
+// getFDCountImproved gets num_fds from /proc/(pid)/fd WITHOUT using the native Readdirnames(),
+// this will skip the step of returning all file names(we don't need) in a dir which takes a lot of memory
+func (p *Probe) getFDCountImproved(pidPath string) int32 {
+	path := filepath.Join(pidPath, "fd")
+
+	if err := p.ensurePathReadable(path); err != nil {
+		return -1
+	}
+
+	d, err := os.Open(path)
+	if err != nil {
+		return -1
+	}
+	defer d.Close()
+
+	b := make([]byte, 8192)
+	count := 0
+
+	for i := 0; ; i++ {
```

Comment:
Do we need to declare this `i` variable here?

Reply:
I don't think so.